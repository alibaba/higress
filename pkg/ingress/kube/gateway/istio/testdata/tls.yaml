apiVersion: gateway.networking.k8s.io/v1alpha2
kind: GatewayClass
metadata:
  name: higress
spec:
  controllerName: higress.io/gateway-controller
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
metadata:
  name: gateway
  namespace: higress-system
spec:
  addresses:
  - value: higress-gateway
    type: Hostname
  gatewayClassName: higress
  listeners:
  - name: passthrough
    port: 34000
    protocol: TLS
    allowedRoutes:
      namespaces:
        from: All
    tls:
      mode: Passthrough
  - name: terminate
    hostname: "domain.example"
    port: 34000
    protocol: HTTPS
    allowedRoutes:
      namespaces:
        from: All
    tls:
      mode: Terminate
      certificateRefs:
      - name: my-cert-http
  - name: terminate-mtls
    hostname: "other.example"
    port: 34000
    protocol: HTTPS
    allowedRoutes:
      namespaces:
        from: All
    tls:
      mode: Terminate
      certificateRefs:
      - name: my-cert-http
      options:
        gateway.higress.io/tls-terminate-mode: MUTUAL
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TLSRoute
metadata:
  name: tls
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  rules:
  - backendRefs:
    - name: httpbin
      port: 443
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: TLSRoute
metadata:
  name: tls-match
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  hostnames:
  - "foo.com"
  rules:
  - backendRefs:
    - name: httpbin-foo
      port: 443
---
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: HTTPRoute
metadata:
  name: http
  namespace: default
spec:
  parentRefs:
  - name: gateway
    namespace: higress-system
  hostnames: ["domain.example"]
  rules:
  - backendRefs:
    - name: httpbin
      port: 80