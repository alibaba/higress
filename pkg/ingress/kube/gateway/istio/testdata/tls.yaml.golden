apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/passthrough.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-passthrough
  namespace: higress-system
spec:
  servers:
  - hosts:
    - '*/*'
    port:
      name: default
      number: 34000
      protocol: TLS
    tls: {}
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/terminate.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-terminate
  namespace: higress-system
spec:
  servers:
  - hosts:
    - '*/domain.example'
    port:
      name: default
      number: 34000
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://higress-system/my-cert-http
      mode: SIMPLE
---
apiVersion: networking.istio.io/v1alpha3
kind: Gateway
metadata:
  annotations:
    internal.istio.io/gateway-service: higress-gateway.higress-system.svc.domain.suffix
    internal.istio.io/parents: Gateway/gateway/terminate-mtls.higress-system
  creationTimestamp: null
  name: gateway-istio-autogenerated-k8s-gateway-terminate-mtls
  namespace: higress-system
spec:
  servers:
  - hosts:
    - '*/other.example'
    port:
      name: default
      number: 34000
      protocol: HTTPS
    tls:
      credentialName: kubernetes-gateway://higress-system/my-cert-http
      mode: MUTUAL
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: HTTPRoute/http.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: http-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-terminate
  hosts:
  - domain.example
  http:
  - name: default/http
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 80
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: TLSRoute/tls-match.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: tls-match-tls-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-passthrough
  hosts:
  - foo.com
  tls:
  - match:
    - sniHosts:
      - foo.com
    route:
    - destination:
        host: httpbin-foo.default.svc.domain.suffix
        port:
          number: 443
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  annotations:
    internal.istio.io/parents: TLSRoute/tls.default
    internal.istio.io/route-semantics: gateway
  creationTimestamp: null
  name: tls-tls-0-istio-autogenerated-k8s-gateway
  namespace: default
spec:
  gateways:
  - higress-system/gateway-istio-autogenerated-k8s-gateway-passthrough
  hosts:
  - '*'
  tls:
  - match:
    - sniHosts:
      - '*'
    route:
    - destination:
        host: httpbin.default.svc.domain.suffix
        port:
          number: 443
---
